sunkit_image/utils/noise.py

```diff
@@ -1,270 +1,306 @@
+
+    Returns
+    -------
+    T : `numpy.ndarray`
+        The new convoluted matrix.
+    """
+    s = np.shape(H)
+    m = int(m)
+    n = int(n)
+    T = np.zeros([(m - s[0] + 1) * (n - s[1] + 1), m * n])
+
+    k = 0
+    for i in range((m - s[0] + 1)):
+        for j in range((n - s[1] + 1)):
+            for p in range(s[0]):
+                T[k, (i + p) * n + j : (i + p) * n + j + 1 + s[1] - 1] = H[p, :]
+
+            k += 1
+    return T
+
+
+def weaktexturemask(img, patchsize, th):
+    """
+    Calculates the weak texture mask.
+    """
+    if len(img.shape) < 3:
+        img = np.expand_dims(img, 2)
+
+    kh = np.expand_dims(np.transpose(np.vstack(np.array([-0.5, 0, 0.5]))), 2)
+    imgh = correlate(img, kh, mode="nearest")
+    imgh = imgh[ :, 1 : imgh.shape[1] - 1, :]
+    imgh = imgh * imgh
+
+    kv = np.expand_dims(np.vstack(np.array([-0.5, 0, 0.5])), 1)
+    imgv = correlate(img, kv, mode="nearest")
+    imgv = imgv[1 : imgv.shape[0] - 1, :, :]
+    imgv = imgv * imgv
+
+    s = img.shape
+    msk = np.zeros_like(img)
+
+    for cha in range(s[2]):
+        m = view_as_windows(img[ :, :, cha], (patchsize, patchsize))
+        m = np.zeros_like(
+            m.reshape(
+                np.int(m.size / patchsize ** 2), patchsize ** 2, order="F"
             ).transpose()
-
-            Xv = view_as_windows(imgv[:, :, cha], (self.patchsize - 2, self.patchsize))
-            Xv = Xv.reshape(
-                np.int(Xv.size / ((self.patchsize - 2) * self.patchsize)),
-                ((self.patchsize - 2) * self.patchsize),
-                order="F",
-            ).transpose()
-
-            Xtr = np.expand_dims(np.sum(np.concatenate((Xh, Xv), axis=0), axis=0), 0)
-
-            p = Xtr < self.th[cha]
-            ind = 0
-
-            for col in range(0, s[1] - self.patchsize + 1):
-                for row in range(0, s[0] - self.patchsize + 1):
-                    if p[:, ind]:
-                        msk[row : row + self.patchsize - 1, col : col + self.patchsize - 1, cha] = 1
-                    ind = ind + 1
-
-        # clean up
-        self.img = np.squeeze(self.img)
-
-        return np.squeeze(msk)
+        )
+
+        Xh = view_as_windows(imgh[:, :, cha], (patchsize, patchsize - 2))
+        Xh = Xh.reshape(
+            np.int(Xh.size / ((patchsize - 2) * patchsize)),
+            ((patchsize - 2) * patchsize),
+            order="F",
+        ).transpose()
+
+        Xv = view_as_windows(imgv[:, :, cha], (patchsize - 2, patchsize))
+        Xv = Xv.reshape(
+            np.int(Xv.size / ((patchsize - 2) * patchsize)),
+            ((patchsize - 2) * patchsize),
+            order="F",
+        ).transpose()
+
+        Xtr = np.expand_dims(np.sum(np.concatenate((Xh, Xv), axis=0), axis=0), 0)
+
+        p = Xtr < th[cha]
+        ind = 0
+
+        for col in range(0, s[1] - patchsize + 1):
+            for row in range(0, s[0] - patchsize + 1):
+                if p[ :, ind]:
```

Comment:
I believe numpy won't like this... what's the boolean value of an array? You should put:
```python
if p[ :, ind].any():
```
or `all()`, depending whether you want one set as `True` or all of them.